fabric/colors.py

```diff
@@ -22,13 +22,23 @@
 If ``bold`` is set to ``True``, the ANSI flag for bolding will be flipped on
 for that particular invocation, which usually shows up as a bold or brighter
 version of the original color on most terminals.
+
+It is now possible to disable color printing completely by setting the environment
+variable FABRIC_DISABLE_COLORS.  The value doesn't matter as long as it is
+Pythonically True.
+
 """
 
+import os
 
 def _wrap_with(code):
 
     def inner(text, bold=False):
         c = code
+
+        if os.environ.get('FABRIC_DISABLE_COLORS') is None:
```

Comment:
Great.  Thanks, Jeff.

I'm an old Perl hacker and was hoping to get a better take on how to
manipulate the fab run environment with command line or config file options
too.  haven't quite figured that out, but an env var is good for batch
processes.

Thanks again.

Eric

On Fri, Apr 8, 2016 at 11:35 PM, Jeff Forcier <notifications@github.com>
wrote:

> In fabric/colors.py
> <https://github.com/fabric/fabric/pull/1271#discussion_r59109647>:
>
> >
> >  def _wrap_with(code):
> >
> >      def inner(text, bold=False):
> >          c = code
> > +
> > +        if os.environ.get('FABRIC_DISABLE_COLORS') is None:
>
> Jeez, yea. Thanks! I didn't even test this by hand :( Just did now to
> confirm. Pushed fix.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/fabric/fabric/pull/1271/files/7cfdc3ce430add37a3206aa959b7ec55cd0e813e#r59109647>
>



-- 
eric berg
eberg@bergbrains.com